include/range/v3/view/generate.hpp

```diff
@@ -39,7 +39,7 @@ namespace ranges
           : view_facade<generate_view<G>, infinite>
         {
         private:
-            friend struct range_access;
+            friend struct ranges::range_access;
```

Comment:
Yowch. The compiler would have caught this bug if these were simply `friend ranges::range_access;` Without the class-key, the friend declaration is not a forward declaration and is ill-formed if the named type is not already declared.